File: misc/benchmarks/programs/ruby/selection-sort.rb
file = File.open("numbers.txt")
str = file.read 
file.close 
arr = str.split(", ").map{ |string| string.to_i }

for i in 0..arr.length-1 do
    # find idx of min value
    min_idx = i
    for j in i..arr.length-1 do 
        if arr[j] < arr[min_idx] then
            min_idx = j
        end

    end

    # swap
    tmp = arr[i]
    arr[i] = arr[min_idx]
    arr[min_idx] = tmp
end

puts "%d - %d" % [ arr[0], arr[-1] ]

File: misc/benchmarks/programs/python/selection-sort.py
file = open('numbers.txt',mode='r')
str = file.read()
file.close()

str = str.split(', ')
arr = list(map(int, str))

for i in range(0, len(arr)):
    min_idx = i 
    for j in range(i+1, len(arr)):
        if arr[j] < arr[min_idx]:
            min_idx = j 

    tmp = arr[i]
    arr[i] = arr[min_idx]
    arr[min_idx] = tmp

print(arr)